src/EventListener/DataContainer/CreateStopElementsSubmitCallbackListener.php

```diff
@@ -0,0 +1,112 @@
+
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\Database;
+use Contao\DataContainer;
+use Contao\Encryption;
+
+/**
+ * @Callback(table="tl_content", target="config.onsubmit")
+ * @Callback(table="tl_form_field", target="config.onsubmit")
+ */
+class CreateStopElementsSubmitCallbackListener
+{
+    public function __invoke(DataContainer $dc): void
+    {
+        $activeRecord = $dc->activeRecord;
+        if (!$activeRecord) {
+            return;
+        }
+
+        if ($activeRecord->type === 'rs_columns_start' || $activeRecord->type === 'rs_column_start') {
+
+            if ($dc->table === 'tl_content') {
+
+                // Find the next columns or column element
+                $nextElement = Database::getInstance()
+                    ->prepare('
+						SELECT type
+						FROM tl_content
+						WHERE pid = ?
+							AND (ptable = ? OR ptable = ?)
+							AND type IN (\'rs_column_start\', \'rs_column_stop\', \'rs_columns_start\', \'rs_columns_stop\')
+							AND sorting > ?
+						ORDER BY sorting ASC
+						LIMIT 1
+					')
+                    ->execute(
+                        $activeRecord->pid,
+                        $activeRecord->ptable ?: 'tl_article',
+                        $activeRecord->ptable === 'tl_article' ? '' : $activeRecord->ptable,
+                        $activeRecord->sorting
+                    );
+
+            }
+            else {
+
+                // Find the next columns or column element
+                $nextElement = Database::getInstance()
+                    ->prepare('
+						SELECT type
+						FROM ' . $dc->table . '
+						WHERE pid = ?
+							AND type IN (\'rs_column_start\', \'rs_column_stop\', \'rs_columns_start\', \'rs_columns_stop\')
+							AND sorting > ?
+						ORDER BY sorting ASC
+						LIMIT 1
+					')
+                    ->execute(
+                        $activeRecord->pid,
+                        $activeRecord->sorting
+                    );
+
+            }
+
+            // Check if a stop element should be created
+            if (
+                !$nextElement->type
+                || ($activeRecord->type === 'rs_columns_start' && $nextElement->type === 'rs_column_stop')
+                || ($activeRecord->type === 'rs_column_start' && (
+                        $nextElement->type === 'rs_column_start' || $nextElement->type === 'rs_columns_stop'
+                    ))
+            ) {
+                $set = array();
+
+                // Get all default values for the new entry
+                foreach ($GLOBALS['TL_DCA'][$dc->table]['fields'] as $field => $config) {
+                    if (array_key_exists('default', $config)) {
+                        $set[$field] = \is_array($config['default']) ? serialize($config['default']) : $config['default'];
+                        if ($GLOBALS['TL_DCA'][$dc->table]['fields'][$field]['eval']['encrypt'] ?? false) {
+                            $set[$field] = Encryption::encrypt($set[$field]);
```

Comment:
The usage of the deprecated `Encryption` class should be refactored - but how?

Reply:
`encrypt` is no longer supported in Contao 5.0, so we should ignore it if the `Encryption` class does not exists I think.

Reply:
I will remove the ecryption!